src/main/scala/com/monsanto/arch/cloudformation/model/resource/Config.scala

```diff
@@ -0,0 +1,235 @@
+    def write(f: SourceOwner) = JsString(f.toString)
+
+    def read(value: JsValue) = {
+      value.toString() match {
+        case "CUSTOM_LAMBDA"  => SourceOwner.CUSTOM_LAMBDA
+        case "AWS"            => SourceOwner.AWS
+      }
+    }
+  }
+}
+
+sealed trait EventSource
+object EventSource extends DefaultJsonProtocol {
+
+  case object `aws.config`       extends EventSource
+
+  implicit val format: JsonFormat[EventSource] = new JsonFormat[EventSource] {
+    def write(f: EventSource) = JsString(f.toString)
+
+    def read(value: JsValue) = {
+      value.toString() match {
+        case "aws.config"      => EventSource.`aws.config`
+      }
+    }
+  }
+}
+
+sealed trait MessageType
+object MessageType extends DefaultJsonProtocol {
+
+  case object ConfigurationItemChangeNotification       extends MessageType
+  case object ConfigurationSnapshotDeliveryCompleted    extends MessageType
+  case object ScheduledNotification                     extends MessageType
+
+  implicit val format: JsonFormat[MessageType] = new JsonFormat[MessageType] {
+    def write(f: MessageType) = JsString(f.toString)
+
+    def read(value: JsValue) = {
+      value.toString() match {
+        case "ConfigurationItemChangeNotification"      => MessageType.ConfigurationItemChangeNotification
+        case "ConfigurationSnapshotDeliveryCompleted"   => MessageType.ConfigurationSnapshotDeliveryCompleted
+        case "ScheduledNotification"                    => MessageType.ScheduledNotification
+      }
+    }
+  }
+}
+
+case class SourceDetail(
+                         EventSource:                Option[EventSource] = None,
+                         MaximumExecutionFrequency:  Option[ExecutionFrequency] = None,
+                         MessageType:                Option[MessageType] = None
+                       )
+object SourceDetail extends DefaultJsonProtocol {
+  implicit val format = jsonFormat3(SourceDetail.apply)
+}
+
+case class Source(
+                   Owner:                Option[SourceOwner] = None,
+                   SourceDetails:        Option[List[SourceDetail]] = None,
+                   SourceIdentifier:     Option[Token[ResourceRef[_]]] = None
+                 )
+object Source extends DefaultJsonProtocol {
+  implicit val format = jsonFormat3(Source.apply)
+}
+
+case class `AWS::Config::ConfigRule`(
+                                      name:                       String,
+                                      ConfigRuleName:             String,
+                                      Description:                String,
+                                      InputParameter:             Map[String, String],
+                                      MaximumExecutionFrequency:  ExecutionFrequency,
+                                      Scope:                      Scope,
+                                      Source:                     Source,
+                                      override val Condition:  Option[ConditionRef] = None
```

Comment:
Would any of these be potential parameters?  i.e., should they be `Token[String]`?  Same question throughout.